ray_lightning/launchers/ray_horovod_launcher.py

```diff
@@ -0,0 +1,251 @@
+from typing import Callable, Any, Optional
+
+import pytorch_lightning as pl
+from pytorch_lightning.strategies.launchers import _Launcher
+from pytorch_lightning.utilities.apply_func import apply_to_collection, \
+    move_data_to_device
+import numpy as np
+import torch
+
+import ray
+from pytorch_lightning.utilities.rank_zero import rank_zero_debug
+from pytorch_lightning.strategies import Strategy
+from ray.util.queue import Queue
+
+from ray_lightning.session import init_session
+from ray_lightning.util import process_results, Unavailable, to_state_stream, \
+    load_state_stream, set_cuda_device_if_used
+
+from ray_lightning.tune import TUNE_INSTALLED, is_session_enabled
+
+try:
+    import horovod.torch as hvd
+    from horovod.ray import RayExecutor
+except (ModuleNotFoundError, ImportError):
+    HOROVOD_AVAILABLE = False
+    RayExecutor = Unavailable
+    hvd = Unavailable
+else:
+    HOROVOD_AVAILABLE = True
+
+from pytorch_lightning.utilities import rank_zero_only
+from ray_lightning.accelerators import \
+    _GPUAccelerator  # noqa: F401
+from ray_lightning.launchers.utils import _RayOutput, get_executable_cls
+
+
+class RayHorovodLauncher(_Launcher):
```

Comment:
Seems like there is a lot of overlap between this launcher and RayLauncher. Would we be able to consolidate a lot of this overlap into a common superclass?

Reply:
Actually, i like this idea!